service/topology_coordinator.cc

```diff
@@ -1836,22 +1788,44 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
                     break;
                 case locator::tablet_transition_stage::end_repair: {
                     if (do_barrier()) {
+                        auto* trinfo = tmap.get_tablet_transition_info(gid.tablet);
+                        if (trinfo && tablet_state.session_id.uuid().is_null()) {
+                            tablet_state.session_id = trinfo->session_id;
+                        }
                         if (action_failed(tablet_state.repair_update_compaction_ctrl)) {
-                            rtlogger.warn("Failed to perform repair_update_compaction_ctrl for tablet repair tablet_id={}", gid);
-                            _tablets.erase(gid);
-                            updates.emplace_back(get_mutation_builder().del_transition(last_token).build());
+                            rtlogger.warn("Failed to perform repair_update_compaction_ctrl for tablet repair tablet_id={} session_id={} nr_retried={}",
+                                    gid, tablet_state.session_id, tablet_state.repair_update_compaction_ctrl_retried);
+                            // So we can retry the repair_update_compaction_ctrl verb
+                            tablet_state.repair_update_compaction_ctrl = std::nullopt;
+                            tablet_state.repair_update_compaction_ctrl_retried++;
                             break;
                         }
                         bool feature = _feature_service.tablet_incremental_repair;
+                        locator::tablet_replica_set replicas;
+                        if (feature) {
+                            for (auto& r : tmap.get_tablet_info(gid.tablet).replicas) {
+                                if (!is_excluded(raft::server_id(r.host.uuid()))) {
```

Comment:
is_excluded() check could be inside parallel_for_each(). This way we avoid building the vector.

Reply:
We will have to build a new vector for the replicas. co_await coroutine::parallel_for_each(tmap.get_tablet_info(gid.tablet).replicas, [] {} ) is too dangerous if the replicas is changed when the loop yields.  It also helps to capture only by value.

Reply:
It is one line now but it could be changed later without considering the possible user after free associated with parallel_for_each later easily e.g, add log after the rpc verb is done. I am not sure why considering safety is now considered as noise, especially, dealing with the tricking parallel_for_each. 

Yes, it is not safe to access sid and gid after the yield point, but it is a local variable, it is much easier to reason about. 

Anyway, I will change to loop the replicas directly and capture by value  in v2.